lib/osf-components/addon/components/storage-provider-manager/osf-storage-manager/component.ts

```diff
@@ -118,4 +123,28 @@ export default class OsfStorageManager extends Component<Args> {
         this.currentPage += 1;
         taskFor(this.getCurrentFolderItems).perform();
     }
+
+    @action
+    selectFile(file: OsfStorageFile, event: PointerEvent) {
+        if (event.shiftKey && this.baseSelectedFile) {
+            const fileIndex = this.displayItems.indexOf(file);
+            const baseIndex = this.displayItems.indexOf(this.baseSelectedFile);
+            const from = Math.min(fileIndex, baseIndex);
+            const to = Math.max(fileIndex, baseIndex) + 1;
+            this.selectedFiles = this.displayItems.slice(from, to);
+        } else {
+            this.baseSelectedFile = file;
+            if (this.selectedFiles.includes(file)) {
+                this.selectedFiles.removeObject(file);
+            } else {
+                this.selectedFiles.pushObject(file);
+            }
+        }
+    }
+
+    @action
+    removeSelectedFiles() {
```

Comment:
Maybe rename this to `deselectFiles()` to avoid confusion with the eventual `deleteSelectedFiles` functionality.